src/Command/CanCleanupVersionCommand.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace Shopware\ServiceBundle\Command;
+
+use Shopware\ServiceBundle\App\AppSelector;
+use Shopware\ServiceBundle\ShopRepository;
+use Symfony\Component\Console\Attribute\AsCommand;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+#[AsCommand(
+    name: 'services:can-cleanup-version',
+    description: 'Checks if any shops are still running on a given version.',
+)]
+class CanCleanupVersionCommand extends Command
+{
+    public function __construct(
+        private ShopRepository $repository,
+        private readonly AppSelector $appSelector,
+    ) {
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->addArgument(
+            'version',
+            InputArgument::REQUIRED,
+            'The version to query.',
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $io = new SymfonyStyle($input, $output);
+
+        $version = $input->getArgument('version');
+        assert(is_string($version));
+
+        if (!$this->appSelector->hasVersion($version)) {
+            $io->error(sprintf('Version "%s" does not exist', $version));
+            return Command::FAILURE;
+        }
+
+        $shopIds = [];
+        foreach ($this->repository->findAll() as $shop) {
+            if ($shop->isShopActive() && $shop->selectedAppVersion === $version) {
+                $shopIds[] = $shop->getShopId();
+            }
+        }
```

Comment:
would it be possible to handle this on the level of the Repository? (e.g. `$this->repository->findActiveShopsForVersion($version)`);